### PR TITLE
Start first incremental sync request early

### PR DESCRIPTION
**Pull request by dbkr**
So it can run while we process our sync data.

### CODE REVIEW

**Comment by dbkr** [comment]
as also discussed IRL, splitting sync state out turned out to be very nontrivial and quite invasive (and didn't feel like much of an improvement). Have fixed the promise chain screwup, could look at rejigging further.

File: src/sync.js [hunk 1]
```diff
@@ -419,27 +419,26 @@ SyncApi.prototype.sync = function() {
...
-                const keepaliveProm = self._startKeepAlives();
-                self._updateSyncState("ERROR", { error: err });
-                return keepaliveProm;
-            }).then(() => {
-                getPushRules();
-            });
+            await savedSyncPromise;
+            const keepaliveProm = self._startKeepAlives();
+            self._updateSyncState("ERROR", { error: err });
+            await keepaliveProm;
```
Comment (lukebarnard1) [inline src/sync.js]: The 6 lines above should probably be factored out from here and from getPushRules

File: src/sync.js [hunk 2]
```diff
@@ -449,42 +448,40 @@ SyncApi.prototype.sync = function() {
...
+
+        if (self._currentSyncRequest === null) {
+            // Send this first sync request here so we can then wait for the saved
+            // sync data to finish processing before we process the results of this one.
+            const qps = self._getSyncParams({ filterId }, savedSyncToken);
+            console.log("Sending first sync request...");
+            self._currentSyncRequest = client._http.authedRequest(
+                undefined, "GET", "/sync", qps, undefined,
+                qps.timeout + BUFFER_PERIOD_MS,
+            );
```
Comment (lukebarnard1) [inline src/sync.js]: I think the above few lines (qps onwards) should be factored out of here and _sync.

File: src/sync.js (hunk 1 above)
Comment (dbkr) [reply src/sync.js]: done, although suggestions welcome on function name

File: src/sync.js (hunk 2 above)
Comment (dbkr) [reply src/sync.js]: done

**Pull request merged**